packages/cf.js/considerations-for-0-0-2.md

```diff
@@ -0,0 +1,13 @@
+# missing features of v0.0.2
+
+- need a way to provide a `timeout` during the install flow
+    - argument to `proposeInstall`?
+    - attribute of the manifest.json
+- currently, the user's address is never being set
+    - provide as an argument: `new Client(nodeProvider, address)`
+    - fetch it from the wallet through an `async init` method?
+- do dapp's specify the deposit? if so, `proposeInstall` and `install` need a deposit arg
+- should the `terms` be set during `install` instead of `proposeInstall`, since we won't know the full limit of the terms until both parties have deposited?
```

Comment:
This would then cause the `terms` to possibly be out of sync between the parties.

In further thinking about the overall flow here, I'm leaning towards more simplicity (at least for the initial version) and making the negotiation the responsibility of the app to handle. 

Simple scenario:
A `installProposal(terms, deposit, ...)` -->  B `ackInstallProposal(terms, deposit, ...)` --> A `install(terms, deposit, ...)`

B doesn't agree:
A `installProposal(terms, deposit, ...)` -->  B `rejectInstallProposal(terms, deposit, ...)`

Upon receiving the rejection, A can either send a new proposal or B can become the initiator.

This makes it such that the `terms` can be mutually agreed on and _then_ having the install proceed, providing better security guarantees.

Reply:
Would this necessitate that A specify the deposit amount for B? As I understand it, `terms.limit` is the sum of their two deposits.

Reply:
That's correct.

In which case, B can effectively send a counter proposal with a deposit amount B is happy with.

Reply:
That makes sense. I wonder if the api should look like this then:

```
proposeInstall(peerAddress, deposit, peerDeposit, terms): Promise<appId>
install(appId)
```

Reply:
Yeah, essentially something like that.

Reply:
Just to confirm, the promise resolves to the app ID and rejects with a `rejectInstall` event type?

Reply:
I suspect that the promise shouldn't resolve with an event type, but that we should instead rely the event observers to handle that. (But, I may be biased. I still don't think we should be return a promise here at all.) What are your thoughts, @mvanderh? Both about returning a `rejectInstall` event and adjusting the api as I described above?